packages/itwinui-css/src/table/base.scss

```diff
@@ -224,7 +224,7 @@
 
     // #region Selection
     &:not([aria-selected='true']) + [aria-selected='true'] .iui-table-cell,
-    &[aria-selected='true']:first-child .iui-table-cell {
+    &[aria-selected='true']:nth-child(2) .iui-table-cell {
```

Comment:
Is this change alright? I had to do it since with the new shadow root div inside table-body, the first selectable element in table body is now the second child.

Without this change, these test images were failing: https://github.com/iTwin/iTwinUI/pull/1725/commits/f6ea60fda7923d4da9336addbb74f64087f5c3ba

Reply:
no, this is because you're not using shadow dom correctly.

the shadow-root should be attached to the table-body element, so that these selectors continue to work.

after attaching the shadow tree, you can add two elements to it:
1. your dummy `<div />` for setting the width
2. a [`<slot />`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/slot) for showing the rows that are its "real" children

Reply:
Ohh, I see. I now attached the shadow-root to the table-body and used `<slot />` to make table-body's children visible even when the shadow-root is attached.

As a result, I also reverted the CSS change from this PR comment. (https://github.com/iTwin/iTwinUI/pull/1725/commits/98bec1d236b93b7cc4e2e91bc2b2f142bbe8a695)